pkg/test/harness.go

```diff
@@ -237,6 +237,17 @@ func (h *Harness) Config() (*rest.Config, error) {
 		return h.config, err
 	}
 
+	// if not the mocked control plane
+	if !h.TestSuite.StartControlPlane {
+		// newly started cluster aren't ready until default service account is ready
+		// fixes: error looking up service account <namespace>/default: serviceaccount "default" not found
+		// no need for this with "inCluster" as we are self evident of it's existence.
```

Comment:
I think it is still possible for the `default/default` service account to be missing when we're running in-cluster (e.g. due to a misconfiguration while we are in another namespace) but it's better to fail straight away on this later rather than uselessly delay here.

```suggestion
		// we avoid this with "inCluster" as the cluster must be already be up since we're running on it
```